compiler-rt/lib/msan/msan.cpp

```diff
@@ -386,44 +420,88 @@ MSAN_MAYBE_STORE_ORIGIN(u16, 2)
 MSAN_MAYBE_STORE_ORIGIN(u32, 4)
 MSAN_MAYBE_STORE_ORIGIN(u64, 8)
 
-void __msan_warning() {
-  GET_CALLER_PC_BP;
-  PrintWarningWithOrigin(pc, bp, 0);
-  if (__msan::flags()->halt_on_error) {
-    if (__msan::flags()->print_stats)
-      ReportStats();
-    Printf("Exiting\n");
-    Die();
+// These macros to reuse the function body are kludgy, but are better than the
+// alternatives:
+// - call a common function: this pollutes the stack traces
```

Comment:
FWIW, if we `always_inline` it should always pollute when we symbolize

Reply:
Polluting is bad, yo (we don't want extra wrappers in the stack trace)